runtime/near-vm-runner/src/tests/fuzzers.rs

```diff
@@ -167,3 +170,31 @@ fn wasmer2_and_wasmtime_agree() {
         assert_eq!(wasmer2, wasmtime);
     });
 }
+
+#[test]
+fn wasmer2_is_reproducible() {
+    bolero::check!().for_each(|data: &[u8]| {
+        if let Ok(module) = ArbitraryModule::arbitrary(&mut arbitrary::Unstructured::new(data)) {
+            let code = ContractCode::new(module.0.module.to_bytes(), None);
+            let config = VMConfig::test();
+            let mut first_hash = None;
+            for _ in 0..3 {
+                let vm = Wasmer2VM::new(config.clone());
+                let prepared_code = match crate::prepare::prepare_contract(code.code(), &config) {
+                    Ok(c) => c,
+                    Err(_) => return,
```

Comment:
Should this be `break` instead?  Isn't `return` going to return from the function and not the closure?

Reply:
looked it up.  the return here will return from the closure not the parent function so all good.

Reply:
Nice keyword to know here is Tennant’s Correspondence Principle: https://without.boats/blog/the-problem-of-effects/